tests/http-proxy.rs

```diff
@@ -0,0 +1,52 @@
+extern crate hyper;
+extern crate alacrity;
+
+use hyper::server::{Server, Request, Response, Handler};
+use hyper::Client;
+use std::io::Read;
+use std::net::SocketAddr;
+
+
+fn with_server<H: Handler + 'static, R>(handle: H, test: &Fn(u16) -> R) -> R {
+    let mut server = Server::http("localhost:0").unwrap().handle(handle).unwrap();
+    let port = server.socket.port();
+
+    // test directly against http server
+    let result_direct = test(port);
+
+    // test against proxy
+    let proxy_addr = "127.0.0.1:8081".to_string();
+    let addr = proxy_addr.parse::<SocketAddr>().unwrap();
+//    alacrity::new_proxy(addr, socket_addr(port));
+//    let result_proxy = test(8081);
```

Comment:
Apologies, I completely missed this reply. In order to test the proxy, we need 3 parts:

1. A server that will listen (block) for new HTTP requests
2. A proxy that will listen (block) for new HTTP requests and forward them to the server
3. A client that will initiate new HTTP requests

Since 1 and 2 are blocking, we need to put those in separate threads. The client request can happen from the main thread.